File: go/1539.go
package main

func findKthPositive(arr []int, k int) int {
	cnt, cur := 0, 0
	for _, v := range arr {
		for v > cur+1 {
			cnt, cur = cnt+1, cur+1
			if cnt == k {
				return cur
			}
		}
		cur++
	}
	return cur + k - cnt
}

File: cpp/1539.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
  int findKthPositive(vector<int> &arr, int k) {
    int cnt = 0, cur = 0;
    for (int i = 0; i < arr.size(); i++) {
      while (arr[i] != cur + 1) {
        ++cur, ++cnt;
        if (cnt == k)
          return cur;
      }
      ++cur;
    }
    return cur + k - cnt;
  }
};
